pkg/services/govmomi/service.go

```diff
@@ -140,6 +140,10 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 		return vm, err
 	}
 
+	if err := vms.reconcileTags(vmCtx); err != nil {
```

Comment:
I think error handling could be slightly improved here instead of reconciling (potentially forever) on any error, including unrecoverable errors.

To quote from the docs:

> AttachMultipleTagsToObject attaches multiple tag IDs to a managed object. This operation is idempotent. If a tag is already attached to the object, then the individual operation is a no-op and no error will be thrown. This operation is not atomic. If the underlying call fails with one or more tags not successfully attached to the managed object reference it might leave the managed object reference in a partially tagged state and needs to be resolved by the caller. In this case BatchErrors is returned and can be used to analyse failure reasons on each failed tag. Specified tagIDs must use URN-notation instead of display names or a generic error will be returned and no tagging operation will be performed. If the managed object reference does not exist a generic 403 Forbidden error will be returned. This operation was added in vSphere API 6.5.

We are working on improved error type handling (for assertions) but the above should give you an impression of different error scenarios.

Reply:
Does it mean that there are no typed errors at the moment?

Reply:
Only partially and the situation is improving as we export more errors. Please see the tests for the method which shows common error scenarios and the errors.